dashboards-observability/public/components/explorer/home_table/history_table.tsx

```diff
@@ -0,0 +1,101 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import React, { useState, useRef, useEffect } from 'react';
+
+import {
+  EuiBasicTable,
+  EuiSpacer,
+  EuiLink,
+} from '@elastic/eui';
+
+
+interface TableData {
+  savedHistory: [];
+  savedQuerySearch: (searchQuery: string, selectedDateRange: [], selectedTimeStamp, selectedFields: []) => void;
+}
+
+export function Table(options: TableData) {
+  const [pageIndex, setPageIndex] = useState(0);
+  const [pageSize, setPageSize] = useState(10);
+  const hisRef = useRef();
+  hisRef.current = pageIndex;
+  const thisRef = useRef();
+  thisRef.current = pageSize;
```

Comment:
Minor: Maybe a more descriptive name? Something like pageSizeRef.

Reply:
yeah I'll change it